src/Operators/vorticity_operators.jl

```diff
@@ -16,27 +22,17 @@
 """
      Γᶠᶠᶜ(i, j, k, grid, u, v)
 
- The vertical circulation associated with horizontal velocities ``u`` and ``v``.
- """
-@inline function Γᶠᶠᶜ(i, j, k, grid::OrthogonalSphericalShellGrid, u, v)
-    # South-west corner
-    if i == 1 && j == 1
-        return Δy_qᶜᶠᶜ(i, j, k, grid, v) - Δx_qᶠᶜᶜ(i, j, k, grid, u) + Δx_qᶠᶜᶜ(i, j-1, k, grid, u)
-
-    # South-east corner
-    elseif i == grid.Nx+1 && j == 1
-        return - Δy_qᶜᶠᶜ(i-1, j, k, grid, v) - Δx_qᶠᶜᶜ(i, j, k, grid, u) + Δx_qᶠᶜᶜ(i, j-1, k, grid, u)
-
-    # North-west corner
-    elseif i == 1 && j == grid.Ny+1
-        return Δy_qᶜᶠᶜ(i, j, k, grid, v) - Δx_qᶠᶜᶜ(i, j, k, grid, u) + Δx_qᶠᶜᶜ(i, j-1, k, grid, u)
-
-    # North-east corner
-    elseif i == grid.Nx+1 && j == grid.Ny+1
-        return - Δy_qᶜᶠᶜ(i-1, j, k, grid, v) - Δx_qᶠᶜᶜ(i, j, k, grid, u) + Δx_qᶠᶜᶜ(i, j-1, k, grid, u)
+The vertical circulation associated with horizontal velocities ``u`` and ``v``.
+"""
 
-    # Not a corner
-    else
-        return δxᶠᵃᵃ(i, j, k, grid, Δy_qᶜᶠᶜ, v) - δyᵃᶠᵃ(i, j, k, grid, Δx_qᶠᶜᶜ, u)
-    end
+@inline function Γᶠᶠᶜ(i, j, k, grid::OrthogonalSphericalShellGrid, u, v)
```

Comment:
do we need a new type? eg ConformalCubedSpherePanel? because now OrthogonalSphericaShellGrid is used more broadly, right?

cc: @simone-silvestri

Reply:
Yes

Reply:
@siddharthabishnu can you deal with this?

Reply:
I can, but as of now, `conformal_cubed_sphere_panel` is a function, not a struct.

Reply:
You should embed the conformal map within the grid:

https://github.com/CliMA/Oceananigans.jl/blob/9c6ed92edf212daaa4c709c4441ce770cdd0bae3/src/Grids/orthogonal_spherical_shell_grid.jl#L46

This is important for two reasons. First, we would like to be able to reproduce the creation of the grid given the grid object. So the parameters that generate the panel should be embedded.

Second, you can use the type of `conformal_mapping` to distinguish between orthogonal grids generated by different mappings. In this way you can tag the grid as being part of the "cubed sphere", thus warranting a different way to compute the circulation.

This was part of changes to orthogonal spherical shell grid a few months ago, concurrent with design of the tripolar grid. The comformal map for the tripolar grid is called `Tripolar`:

https://github.com/CliMA/OrthogonalSphericalShellGrids.jl/blob/379d0de985a0927c026e6c3e43b2ebd163e054aa/src/tripolar_grid.jl#L325

I suggest working first on generating the grid in a clean and understandable way, designing the conformal mapping type. Merge that to main with tests. Next, use that PR to dispatch on the special grid type for computing circulation at the panel corners.

Reply:
@glwagner, what do you mean embed the map within the grid?

Reply:
However, at the moment we are only dispatching on that parameter being `Something` or `Nothing`. You are suggesting we create a type for the conformal map and dispatch over that type, right? (And similarly, we'll create a type for the tripolar map over the OrthogonalSphericalShellGrids.jl package..)

Reply:
> what do you mean "embed the map within the grid"? it's already a field and a parameter of the OrthogonalSphericalShellGrid type, no?

Then why can't you simply dispatch on this special kind of `OrthogonalSphericalShellGrid` to compute the cubed sphere circulation?

Reply:
Yeap, we should do that.

Reply:
> And similarly, we'll create a type for the tripolar map over the OrthogonalSphericalShellGrids.jl package

This already exists

Reply:
@glwagner, @navidcy please check out PR #3832.